File: API/Models/Category.cs
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System;

namespace QuizAppApi.Models
{
    public class Category : BaseEntity
    {
        public string Title { get; set; }
        public List<ChallengeCategory> ChallengeCategories { get; set; }
        public List<CategoryQuestion> CategoryQuestions { get; set; }
    }
}

File: src/app/models/category.ts
import { ChallengeCategory } from "./challengecategory";
import { CategoryQuestion } from "./categoryquestion";
import { environment } from "../../environments/environment";

export class Category {
    constructor() {
        this.id = environment.emptyGuid;
    }
    id: string;
    title: string;
    creationDate: Date;
    deletionDate: Date;
    isDeleted: boolean;
    challengeCategories: ChallengeCategory[];
    categoryQuestions: CategoryQuestion[];
}
